ext/spl/spl_iterators.stub.php

```diff
@@ -0,0 +1,475 @@
+    /** @return bool */
+    public function hasChildren() {}
+
+    /** @return RecursiveFilterIterator|null */
+    public function getChildren() {}
+}
+
+class ParentIterator extends RecursiveFilterIterator
+{
+    public function __construct(RecursiveIterator $iterator) {}
+
+    /** @return bool */
+    public function accept() {}
+}
+
+interface SeekableIterator extends Iterator
+{
+    /** @return void */
+    public function seek(int $position);
+}
+
+class LimitIterator extends IteratorIterator
+{
+    public function __construct(Iterator $iterator, int $offset = 0, int $count = -1) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return int */
+    public function seek(int $position) {}
+
+    /** @return int */
+    public function getPosition() {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+}
+
+class CachingIterator extends IteratorIterator implements ArrayAccess, Countable
+{
+    public function __construct(Iterator $iterator, int $flags = self::CALL_TOSTRING) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return bool */
+    public function hasNext() {}
+
+    public function __toString(): string {}
+
+    /** @return Iterator|null */
+    public function getInnerIterator() {}
+
+    /** @return int */
+    public function getFlags() {}
+
+    /** @return void */
+    public function setFlags(int $flags) {}
+
+    /**
+     * @param string $index
+     * @return mixed
+     */
+    public function offsetGet($index) {}
+
+    /**
+     * @param string $index
+     * @param mixed $value
+     * @return void
+     */
+    public function offsetSet($index, $value) {}
+
+    /**
+     * @param string $index
+     * @return void
+     */
+    public function offsetUnset($index) {}
+
+    /**
+     * @param string $index
+     * @return bool
+     */
+    public function offsetExists($index) {}
+
+    /** @return array */
+    public function getCache() {}
+
+    /** @return int */
+    public function count() {}
+}
+
+class RecursiveCachingIterator extends CachingIterator implements RecursiveIterator
+{
+    public function __construct(Iterator $iterator, int $flags = self::CALL_TOSTRING) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
```

Comment:
It's not necessary to list the methods inherited from CachingIterator here.

Reply:
I use PHPStorm stubs as a base ^^